components/hmpo-character-count/spec.macro.js

```diff
@@ -0,0 +1,167 @@
+'use strict';
+
+describe('hmpoCharacterCount', () => {
+    let locals;
+
+    beforeEach(() => {
+        locals = {
+            options: {
+                fields: {
+                    'my-input': {
+                        validate: 'required'
+                    }
+                }
+            },
+            values: {
+                'my-input': 'abc123'
+            }
+        };
+    });
+
+    it('renders with id', () => {
+        const $ = render({ component: 'hmpoCharacterCount', params: { id: 'my-input' }, ctx: true }, locals);
+
+        const $component = $('.govuk-textarea');
+        expect($component.attr('id')).to.equal('my-input');
+    });
+
+    it('renders with label and hint', () => {
+        const $ = render({ component: 'hmpoCharacterCount', params: { id: 'my-input' }, ctx: true }, locals);
+
+        const $label = $('.govuk-label');
+        expect($label.text().trim()).to.equal('[fields.my-input.label]');
+        expect($label.attr('id')).to.equal('my-input-label');
+        const $hint = $('.govuk-hint');
+        expect($hint.text().trim()).to.equal('[fields.my-input.hint]');
```

Comment:
Could we restructure the tests a bit? 

Logic looks good to me but I think the format could be better. See example below

```
const $ = render({ component: 'hmpoCharacterCount', params: { id: 'my-input' }, ctx: true }, locals);
const $hint = $('.govuk-hint');
const $label = $('.govuk-label');

expect($label.text().trim()).to.equal('[fields.my-input.label]');
expect($label.attr('id')).to.equal('my-input-label');
expect($hint.text().trim()).to.equal('[fields.my-input.hint]');
```